packages/web/src/pages/settings-page/components/desktop/DeveloperApps/EditAppPage.tsx

```diff
@@ -0,0 +1,211 @@
+
+const { getUserId } = accountSelectors
+
+type EditAppPageProps = CreateAppPageProps
+
+type DeveloperAppValues = z.input<typeof developerAppEditSchema>
+
+const messages = {
+  appNameLabel: 'App Name',
+  imageUrlLabel: 'App Icon URL',
+  apiKey: 'api key',
+  copyApiKeyLabel: 'copy api key',
+  copied: 'Copied!',
+  goBack: 'Back to Your Apps',
+  back: 'Back',
+  save: 'Save Changes',
+  saving: 'Saving',
+  miscError: 'Sorry, something went wrong. Please try again later.'
+}
+
+const ImageField = ({ name }: { name: string }) => {
+  const [{ value }] = useField(name)
+  return value
+    ? <img src={value} height='136' width='136' />
+    : <Flex
+        w='100%'
+        justifyContent='center'
+        alignItems='center'
+        borderRadius='l'
+        css={{ backgroundColor: 'var(--harmony-n-200)' }}
+      >
+        <IconEmbed
+          color='subdued'
+          css={{ width: '75px', height: '75px' }}
+        />
+      </Flex>
+}
+
+export const EditAppPage = (props: EditAppPageProps) => {
+  const { params, setPage } = props
+  const userId = useSelector(getUserId) as number
+  const { name, description, apiKey, imageUrl } = params || {}
+
+  const record = useRecord()
+
+  const [editDeveloperApp, result] = useEditDeveloperApp()
+  const [submitError, setSubmitError] = useState<string | null>(null)
+
+  const { status, data, errorMessage } = result
+
+  useEffect(() => {
+    if (status === Status.SUCCESS) {
+      setPage(CreateAppsPages.YOUR_APPS)
+      record(
+        make(Name.DEVELOPER_APP_EDIT_SUCCESS, {
+          name: name || '',
+          apiKey: apiKey || ''
+        })
+      )
+    }
+  }, [data, record, setPage, status])
+
+  useEffect(() => {
+    if (status === Status.ERROR) {
+      setSubmitError(messages.miscError)
+      record(
+        make(Name.DEVELOPER_APP_EDIT_ERROR, {
+          error: errorMessage
+        })
+      )
+    }
+  }, [errorMessage, record, status])
+
+  const handleSubmit = useCallback(
+    (values: DeveloperAppValues) => {
+      setSubmitError(null)
+      record(
+        make(Name.DEVELOPER_APP_EDIT_SUBMIT, {
+          name: values.name,
+          description: values.description
+        })
+      )
+      editDeveloperApp(values)
+    },
+    [editDeveloperApp, record]
+  )
+
+  const initialValues: DeveloperAppValues = {
+    userId,
+    apiKey: apiKey || '',
+    name: name || '',
+    description,
+    imageUrl
+  }
+
+  const isSubmitting = status !== Status.IDLE && status !== Status.ERROR
+
+  const copyApiKey = useCallback(() => {
+    if (!apiKey) return
+    copyToClipboard(apiKey)
+  }, [apiKey])
+
+  if (!params) return null
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      onSubmit={handleSubmit}
+      validationSchema={toFormikValidationSchema(developerAppEditSchema)}
+    >
+      {() => (
```

Comment:
Is the function child component necessary if we aren't actually using the arguments to the render function? Feels like extra rendering indirection for no benefit.